File: app/src/main/java/com/permissionnanny/dagger/ContextComponent.java
package com.permissionnanny.dagger;

import android.content.Context;
import com.permissionnanny.ClientPermissionManifestReceiver;
import com.permissionnanny.ClientRequestReceiver;
import com.permissionnanny.ConfirmRequestBinder;
import com.permissionnanny.ProxyService;
import com.permissionnanny.UninstallReceiver;
import com.permissionnanny.dagger.ContextComponent.ContextScope;
import com.permissionnanny.missioncontrol.AppControlBinder;
import dagger.Component;
import javax.inject.Scope;

/**
 *
 */
@ContextScope
@Component(modules = {ContextModule.class}, dependencies = AppComponent.class)
public interface ContextComponent {

    @Scope
    @interface ContextScope {}

    Context context();

    void inject(ProxyService victim);

    void inject(ClientRequestReceiver victom);

    void inject(ClientPermissionManifestReceiver victim);

    void inject(UninstallReceiver victim);

    void inject(ConfirmRequestBinder victim);

    void inject(AppControlBinder victim);
}

File: app/src/main/kotlin/com/permissionnanny/dagger/ContextComponent.kt
package com.permissionnanny.dagger

import android.content.Context
import com.permissionnanny.*
import com.permissionnanny.dagger.ContextComponent.ContextScope
import dagger.Component
import javax.inject.Scope

/**

 */
@ContextScope
@Component(modules = arrayOf(ContextModule::class), dependencies = arrayOf(AppComponent::class))
interface ContextComponent {

    @Scope
    annotation class ContextScope

    fun context(): Context

    fun inject(victim: ProxyService)

    fun inject(victim: ClientRequestReceiver)

    fun inject(victim: ClientPermissionManifestReceiver)

    fun inject(victim: UninstallReceiver)

    fun inject(victim: ConfirmRequestBinder)
}
